_overviews/scala3-scaladoc/snippet-compiler.md

```diff
@@ -0,0 +1,226 @@
+
+To configure it, you need to add a `sc-name` argument directly in the snippet that you want to include:
+
+```` ```scala sc-name:<snippet-name> ````
+
+where `snippet-name` should be unique in file scope and cannot contain whitespaces and commas.
+
+Then, you can add a `sc-compile-with` argument directly in the snippet that should contain include:
+```` ```scala sc-compile-with:<snippet-name>(,<snippet-name>)+ ````
+
+where `snippet-name` is the name of snippet that should be included.
+
+After configuring this feature in our example, the code looks like this:
+
+![]({{ site.baseurl }}/resources/images/scala3/scaladoc/documentation-snippet2.png)
+
+And the output looks like this:
+
+![]({{ site.baseurl }}/resources/images/scala3/scaladoc/snippet-includes.png)
+
+You can specify more than one include. The order in which they are specified defines the order of inclusion. 
+
+**Warning**: you can only include snippets that are defined above the target snippet.
+
+## Advanced configuration
+
+Often turning on snippet valiation for all snippets is not enough as the usecases can be more sophisticated. We prepared our tool for such situations to allow user adjust it to his needs.
+
+### Available flags
+
+Snippet compiler exposes three flags that change its behavior:
+- `compile` - turns on snippet checking
+- `nocompile` - turns off snippet checking
+- `fail` - turns on snippet checking with assertion that snippets don't compile
+
+### Path-based settings
+
+Instead of setting one flag to all snippets in project, it can be set for certain path only. It can be done by adding `<path>=` prefix before flag.
+
+Example:
+
+`-snippet-compiler:docs=compile` - sets `compile` flag for snippets in `docs` file. If `docs` is a directory, the flag is set for all files inside `docs`.
+
+Additionally, `-snippet-compiler` option can be supplied by more than one setting. Settings are delimited by commas.
+
+Example:
+```
+-snippet-compiler:docs=compile,library/src=compile,library/src/scala/quoted=nocompile,library/src/scala/compiletime=fail
+```
+Flags are chosen by longest prefix match so we can define a general setting and then change behavior for more specific paths.
+```
+-snippet-compiler:compile,library/src/scala/quoted=nocompile,library/src/scala/compiletime=fail 
+```
+Flag without path prefix is treated as default.
+
+### Override directly in the snippet
+
+CLI arguments are a good mechanism for setting flags for certain files. However, it cannot be used to configure snippet compiler for certain snippet. For example an author wants to write one snippet that should fail and few snippets that compile. We took that under consideration and added a feature to override settings directly in the snippet. These arguments are located in snippet info part:
```

Comment:
```suggestion
CLI arguments are a good mechanism for setting flags for certain files. However, this approach can’t be used to configure the snippet compiler for specific snippets. For example, an author wants to write one snippet that should fail, and other snippets that should compile. Again we took this under consideration, and added a feature to override settings directly inside snippets. These arguments are located in the snippet info part:
```

Is there a more proper name for the “snippet info part”? Earlier we refer to this area as a flag, but I’m not sure if that’s correct here.